site_scons/buildutils.py

```diff
@@ -308,89 +405,85 @@ def compareProfiles(env, ref_file, sample_file):
-
-        comp = np.interp(reference[0], sample[0], sample[i])
-        for j in range(nTimes):
-            a = reference[i,j]
-            b = comp[j]
-            abserr = abs(a-b)
-            relerr = abs(a-b) / (scale + atol)
-
-            # error that can be accounted for by shifting the profile along
-            # the time / spatial coordinate
-            xerr = abserr / (abs(slope[j]) + atol)
-
-            if abserr > atol and relerr > rtol and xerr > xtol:
-                bad.append((reference[0][j], i, a, b, abserr, relerr, xerr))
+    template = '{0:9.4e}  {1:5d}  {2:14.7e}  {3:14.7e}  {4:9.3e}  {5:9.3e}  {6:9.3e}'
+    ref_ptp = reference.ptp(axis=1)
+    ref_max = np.abs(reference).max(axis=1)
+    sample_ptp = sample.ptp(axis=1)
+    sample_max = np.abs(sample).max(axis=1)
+    scale = np.maximum(
+        np.maximum(ref_ptp[1:], ref_max[1:]),
+        np.maximum(sample_ptp[1:], sample_max[1:])
+    ).reshape(n_vars-1, -1)
+    ref_diff = np.diff(reference)
+    slope = ref_diff[1:, :] / ref_diff[0, :] * ref_ptp[0]
+    slope = np.hstack((np.zeros((n_vars - 1, 1)), slope))
+    comp = np.zeros((n_vars - 1, n_times))
+    for i, row in enumerate(sample[1:]):
+        comp[i, :] = np.interp(reference[0, :], sample[0, :], row)
+
+    abserr = np.abs(reference[1:] - comp)
+    relerr = abserr / (scale + atol)
+    # error that can be accounted for by shifting the profile along
+    # the time / spatial coordinate
+    xerr = abserr / (np.abs(slope) + atol)
+    if np.any(abserr > atol) and np.any(relerr > rtol) and np.any(xerr > xtol):
+        it = np.nditer((abserr, relerr, xerr), flags=["multi_index"])
+        for a, r, x in it:
+            i, j = it.multi_index
+            bad.append((reference[0, j], i, reference[i, j], comp[i, j], a, r, x))
 
     footer = []
     maxrows = 10
     if len(bad) > maxrows:
         bad.sort(key=lambda row: -row[5])
-        footer += ['Plus {0} more points exceeding error thresholds.'.format(len(bad)-maxrows)]
+        footer += [f"Plus {len(bad) - maxrows} more points exceeding error thresholds."]
         bad = bad[:maxrows]
 
     if bad:
-        header = ['Failed series comparisons:',
-                  'coordinate  comp.  reference val    test value    abs. err   rel. err    pos. err',
-                  '----------  ---   --------------  --------------  ---------  ---------  ---------']
-        print('\n'.join(header + [template.format(*row) for row in bad] + footer))
-        return 1
-    else:
-        return 0
-
-
-def getPrecision(x):
-    """
-    Return the number corresponding to the least significant digit of
-    the number represented by the string 'x'.
-    """
-    x = x.lower()
-    # Patterns to consider:
-    # 123
-    # 123.45
-    # 123.45e6
-    # 123e4
-    if 'e' in x:
-        x, exponent = x.split('e')
-        exponent = int(exponent)
+        header = ["Failed series comparisons:"]
+        header.append("coordinate  comp.  reference val    test value    "
```

Comment:
here I'd almost recommend an exception to the line length, or put things on it's own line with `"""` (i.e. the old formatting was clearer)

Reply:
I reformatted this, hopefully it's clearer now? The advantage is that the field width is a little more obvious now that I'm using `str.format`, the disadvantage is, why are you using `str.format` with string literal arguments? Ah well.

Reply:
Somewhat yes, although the original formatting wasn’t too bad either. Your call ...